js/src/components/tree-select-control/index.js

```diff
@@ -163,52 +271,107 @@ const TreeSelectControl = ( {
 
 		let filteredTreeOptions = cloneDeep( treeOptions );
 
-		if ( filter ) {
-			filteredTreeOptions = filteredTreeOptions.filter( filterOption );
-		}
-
+		filteredTreeOptions = filteredTreeOptions.filter( filterOption );
 		filteredOptionsCache.current[ filter ] = filteredTreeOptions;
 		setFilteredOptions( filteredTreeOptions );
-	}, [ treeOptions, filter ] );
+	}, [ treeOptions, filter, getOption ] );
 
 	const onKeyDown = ( event ) => {
-		if ( inputControlValue ) return;
+		if ( ESCAPE === event.key ) {
+			setTreeVisible( false );
+		}
 
-		if ( BACKSPACE === event.key ) {
-			onChange( value.slice( 0, -1 ) );
+		if ( ENTER === event.key ) {
+			setTreeVisible( true );
 			event.preventDefault();
 		}
 
-		if ( ESCAPE === event.key ) {
-			setTreeVisible( false );
+		if ( inputControlValue ) return;
```

Comment:
> Navigation is disabled when searching.

Will this be added in the subsequent PR or be a future roadmap?

Reply:
Yes, I'm working already on that (in future PR). However, we will need to iterate manually over the tree since we don't know which nodes are present or not. But I don't think it will be problematic since after the search we should have fewer nodes.